File: articles/calculus-for-systems-modeling/c/calculus_logistic_model.c
#include <stdio.h>

int main(void) {
    const int steps = 300;
    const double dt = 0.1;
    const double rate = 0.20;
    const double capacity = 100.0;

    double state[steps];
    state[0] = 10.0;

    for (int i = 1; i < steps; i++) {
        double derivative = rate * state[i - 1] * (1.0 - state[i - 1] / capacity);
        state[i] = state[i - 1] + derivative * dt;
    }

    printf("Final state: %.6f\n", state[steps - 1]);

    return 0;
}

File: articles/calculus-for-systems-modeling/go/calculus_logistic_model.go
package main

import "fmt"

func simulateLogistic(initialState, rate, capacity, dt float64, steps int) []float64 {
	state := make([]float64, steps)
	state[0] = initialState

	for i := 1; i < steps; i++ {
		derivative := rate * state[i-1] * (1.0 - state[i-1]/capacity)
		state[i] = state[i-1] + derivative*dt
	}

	return state
}

func main() {
	state := simulateLogistic(10.0, 0.20, 100.0, 0.1, 300)
	fmt.Printf("Final state: %.6f\n", state[len(state)-1])
}
